sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/executable/SonarTSCoreBundle.java

```diff
@@ -87,16 +95,28 @@ private void deploy() {
    * Builds command to run rules and calculate metrics with tsrunner
    */
   @Override
-  public SonarTSRunnerCommand getSonarTsRunnerCommand(String tsconfigPath, Iterable<InputFile> inputFiles, TypeScriptRules typeScriptRules) {
-    String increaseMemory = "--max-old-space-size=" + NODE_PROCESS_MEMORY;
-    SonarTSRunnerCommand runnerCommand = new SonarTSRunnerCommand(inputFiles, getNodeExecutable(), increaseMemory, this.tsMetricsExecutable.getAbsolutePath());
-    runnerCommand.setTsConfigPath(tsconfigPath);
+  public SonarTSCommand getSonarTsRunnerCommand() {
+    return getCommand(sonartsRunnerExecutable);
+  }
+
+  @Override
+  public String buildRequest(String tsconfigPath, Iterable<InputFile> inputFiles, TypeScriptRules typeScriptRules) {
+    SonarTSRequest request = new SonarTSRequest();
+    request.filepaths = StreamSupport.stream(inputFiles.spliterator(), false).map(InputFile::absolutePath).toArray(String[]::new);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "absolutePath"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=sonarlint-integration-poc&issues=AWDb-EgwbBlXo5JPxazf&open=AWDb-EgwbBlXo5JPxazf)

[ISSUE_KEY]: AWDb-EgwbBlXo5JPxazf